nbextensions/appCell2/widgets/tabs/resultsViewer.js

```diff
@@ -15,119 +15,170 @@ define([
 
-            const { jobState } = arg;
+            const layout = div(
+                {
+                    class: 'kb-app-results-tab',
+                },
+                [
+                    ui.buildCollapsiblePanel({
+                        title: 'Results',
+                        name: 'results',
+                        hidden: true,
+                        type: 'default',
+                        classes: ['kb-panel-results'],
+                    }),
+                    div({ dataElement: 'report' }),
+                    div({ dataElement: 'next-steps' }),
+                ]
+            );
+            container.innerHTML = layout;
 
-            return Promise.try(() => {
-                const layout = div(
-                    {
-                        style: {
-                            overflowX: 'auto',
-                            maxWidth: 'inherit',
-                        },
-                    },
-                    [
-                        ui.buildCollapsiblePanel({
-                            title: 'Results',
-                            name: 'results',
-                            hidden: true,
-                            type: 'default',
-                            classes: ['kb-panel-results'],
-                        }),
-                        div({ dataElement: 'report' }),
-                        div({ dataElement: 'next-steps' }),
-                    ]
-                );
-                container.innerHTML = layout;
+            return Promise.all([showResults(arg.jobState, arg.isParentJob), showNextApps()]);
+        }
 
-                // If there's a "report_name" key in the results, load and show the report.
-                const result = model.getItem('exec.outputWidgetInfo');
-                if (arg.isParentJob && result && result.params && result.params.report_name) {
-                    renderReportView(result.params);
-                } else if (
-                    jobState.widget_info &&
-                    jobState.widget_info.params &&
-                    jobState.widget_info.params.report_name
-                ) {
-                    // do report widget.
-                    renderReportView(jobState.widget_info.params);
-                } else {
-                    ui.getElement('results').classList.remove('hidden');
-                    ui.setContent(
-                        'results.body',
-                        ui.buildPresentableJson(jobState.job_output.result)
-                    );
-                }
+        /**
+         * Shows the results of a job run. This returns a Promise that resolves when
+         * the rendering is done.
+         * @param {Object} jobState - object describing the state of a job
+         * @param {boolean} isParentJob - true if this job is a parent of a batch job
+         * @returns
+         */
+        function showResults(jobState, isParentJob) {
+            // there's a couple of ways to find report parameters.
```

Comment:
The guts of this are mostly unchanged from what used to be part of the `start` function. Just moved to `showResults` instead.